module/data/item/templates/class.mjs

```diff
@@ -171,6 +171,17 @@ export default class ClassTemplate extends ItemDataModel.mixin(
     await this.parentActor.createEmbeddedDocuments( "ActiveEffect", effects );
     // TODO: activate permanent effects immediately
 
+    // increase resource step of the discipline
+    const highestClass = this.parentActor.highestDiscipline;
+    
```

Comment:
I would call it `highestDiscipline`, since it's never gonna be a questor class. Also, since it's only used once, you could just call the getter down in the comparison itself.

Reply:
good point